src/http.zig

```diff
@@ -1084,7 +1153,14 @@ pub const HTTPThread = struct {
                     requested_config.deinit();
                     bun.default_allocator.destroy(requested_config);
                     bun.default_allocator.destroy(custom_context);
-                    return err;
+
+                    // TODO: these error names reach js. figure out how they should be handled
+                    return switch (err) {
```

Comment:
Because it's a little out of scope for this pr, I return `error.FailedToOpenSocket` for each error. Open to changing this